src/components/tables/DataTable.vue

```diff
@@ -198,16 +208,24 @@ const {
   singleExpand,
   stickyOffset,
   stickyHeader,
+  group,
+  collapsed,
 } = toRefs(props);
 
 const css = useCssModule();
+const slots = useSlots();
 const { stick, table, tableScroller } = useStickyTableHeader(
   stickyOffset,
   stickyHeader,
 );
 const tableDefaults = useTable();
+
+const headerSlots = computed(() =>
+  Object.keys(slots).filter((slotName) => slotName.startsWith('header.')),
+);
+
 const globalItemsPerPageSettings = computed(() => {
-  if (props.globalItemsPerPage !== undefined) {
+  if (typeof props.globalItemsPerPage !== 'undefined') {
```

Comment:
Is there any specific reason why we need to do type-checking here instead of value checking?

Reply:
<img width="728" alt="image" src="https://github.com/rotki/ui-library-compat/assets/5625068/c1c6918b-874c-49a9-8bbe-f067eb6a5379">

Condition is always true since types 'boolean' and 'undefined' have no overlap

that's ^ the warning that prompted this change.